shared/MAPL_ShmemMod.F90

```diff
@@ -198,6 +198,8 @@ subroutine MAPL_InitializeShmem(rc)
 
       allocate(Segs(CHUNK),stat=STATUS)
       _ASSERT(STATUS==0,'needs informative message')
+      Segs(:)%shmid = -1
+      Segs(:)%addr=C_NULL_PTR
```

Comment:
Not worth the change at this point, but this could be more clearly implemented as:
```
segs(:) = SegmentT(-1_C_INT, C_NULL_PTR)
```
(esp not worth it because of how ugly the "-1" must be declared.)